asf_core_data/pipeline/data_joining/merge_proc_datasets.py

```diff
@@ -206,40 +236,47 @@ def merging_pipeline(
     """
 
     # Load the processed EPC data (not deduplicated)
-    prep_epc = load_preprocessed_epc_data(
+    merged_data = load_preprocessed_epc_data(
         data_path=path_to_data,
         version="preprocessed",
         batch="newest",
         usecols=epc_usecols,
         verbose=verbose,
     )
 
-    # Add coordinates for EPC data
-    prep_epc = feature_engineering.get_postcode_coordinates(
-        prep_epc, postcode_field_name="POSTCODE"
-    )
-
     # Add more precise estimations for heat pump installation dates via MCS data
-    epc_with_MCS_dates = install_date_computation.compute_hp_install_date(
-        prep_epc, verbose=verbose
+    merged_data = install_date_computation.compute_hp_install_date(
+        merged_data, verbose=verbose
     )
 
     # Merge EPC with MCS installations
-    epc_mcs_insts = add_mcs_installations_data(
-        epc_with_MCS_dates, usecols=mcs_installations_usecols, verbose=verbose
+    merged_data = add_mcs_installations_data(
+        merged_data, usecols=mcs_installations_usecols, verbose=verbose
     )
 
     # Merge EPC/MCS with MCS installers
-    epc_mcs_complete = add_mcs_installer_data(
-        epc_mcs_insts, usecols=mcs_installers_usecols
+    merged_data = add_mcs_installer_data(merged_data, usecols=mcs_installers_usecols)
+
+    # Add coordinates for EPC data
+    merged_data = feature_engineering.get_postcode_coordinates(
+        merged_data, postcode_field_name="POSTCODE"
     )
 
     today = date.today().strftime("%y%m%d")
 
+    # Replacing all types of missing with NaN
+    merged_data.replace("Unknown", np.nan, inplace=True)
+    merged_data.replace("unknown", np.nan, inplace=True)
+    merged_data.replace("Undefined", np.nan, inplace=True)
+    merged_data.replace("Unspecified", np.nan, inplace=True)
+    merged_data.replace("", np.nan, inplace=True)
```

Comment:
Maybe better to have a list with all "unknowns" and then iterate over it. In feature_cleaning_utils.py, we have a list called invalid_values. Maybe you could iterate over all of them, plus the ones you have here.

```
for value in invalid_values: merged_data.replace(value, np.nan, inplace=True)
```